drivers/syslog/syslog_rpmsg_server.c

```diff
@@ -64,11 +70,77 @@ static void syslog_rpmsg_ns_unbind(FAR struct rpmsg_endpoint *ept);
 static int  syslog_rpmsg_ept_cb(FAR struct rpmsg_endpoint *ept,
                                 FAR void *data, size_t len, uint32_t src,
                                 FAR void *priv_);
+#ifdef CONFIG_SYSLOG_RPMSG_SERVER_CHARDEV
+static int syslog_rpmsg_file_ioctl(FAR struct file *filep, int cmd,
+                                   unsigned long arg);
+#endif
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+#ifdef CONFIG_SYSLOG_RPMSG_SERVER_CHARDEV
+static struct list_node g_list = LIST_INITIAL_VALUE(g_list);
+static mutex_t g_lock = NXMUTEX_INITIALIZER;
+
+static const struct file_operations g_syslog_rpmsg_fops =
+{
+  NULL,                    /* open */
+  NULL,                    /* close */
+  NULL,                    /* read */
+  NULL,                    /* write */
+  NULL,                    /* seek */
+  syslog_rpmsg_file_ioctl, /* ioctl */
+  NULL                     /* poll */
+#ifndef CONFIG_DISABLE_PSEUDOFS_OPERATIONS
+  , NULL                   /* unlink */
+#endif
+};
 
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
 
+static int syslog_rpmsg_file_ioctl(FAR struct file *filep, int cmd,
+                                   unsigned long arg)
+{
+  FAR struct syslog_rpmsg_server_s *priv;
+  struct syslog_rpmsg_sync_s msg;
+  sem_t sem;
+  int ret = 0;
+
+  if (cmd != FIOC_DUMP)
+    {
+      return -ENOTTY;
+    }
+
+  nxsem_init(&sem, 0, 0);
+  nxsem_set_protocol(&sem, SEM_PRIO_NONE);
+
+  nxmutex_lock(&g_lock);
+  list_for_every_entry(&g_list, priv, struct syslog_rpmsg_server_s, node)
+    {
+      msg.cookie = (uint64_t)(uintptr_t)&sem;
+      msg.header.command = SYSLOG_RPMSG_SYNC;
+      ret = rpmsg_send(&priv->ept, &msg, sizeof(msg));
+      if (ret < 0)
+        {
+          continue;
+        }
+
+      ret = rpmsg_wait(&priv->ept, &sem);
+      if (ret < 0)
+        {
+          continue;
+        }
+    }
+
+  nxmutex_unlock(&g_lock);
+  nxsem_destroy(&sem);
+  return ret;
```

Comment:
```suggestion
  return OK;
```

Reply:
Done. update in https://github.com/apache/incubator-nuttx/pull/6655. Thanks.